packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts

```diff
@@ -28,6 +29,23 @@ function patchedAddEventListener(
         // @ts-ignore type-mismatch
         return addCustomElementEventListener.apply(this, arguments);
     }
+
+    // TODO [#4011]: Due to a bug and how the synthetic shadow polyfill currently caches the composed path of events,
+    // events originating from native shadow components can incorrectly use a cached composed path from
+    // a context it shouldn't have access to. One solution is to avoid patching the addEventListener
+    // for event targets inside a component using native shadow. This avoids applying our synthetic
+    // shadow polyfill behavior unnecessarily.
+    //
+    // Open Questions:
+    //  - Some targets are not instances of Node (e.g., an instance of XMLHttpRequest)
+    //    Do we need fallback behavior for non-Node instances?
```

Comment:
I think it makes sense to have an `else` here that also bypasses the wrapper for non-Node instances because they will not be a part of the DOM and will not require access validation. We do something similar where [we return an empty array for the composed path](https://github.com/salesforce/lwc/blob/582ad5264a8cc4cedb0aaedc8ec1006661c43c1a/packages/%40lwc/synthetic-shadow/src/polyfills/event/polyfill.ts#L91), which is what we use to determine access.